File: src/henaxel/workbench/Workbench.java
package henaxel.workbench;

import javax.swing.*;
import java.awt.*;

public class Workbench extends JFrame {
    JComponent workSpace;

    public Workbench() {
        getContentPane().setLayout(new BorderLayout());

        setDefaultCloseOperation(WindowConstants.EXIT_ON_CLOSE);
        setResizable(true);
        setSize(new Dimension((int)(Toolkit.getDefaultToolkit().getScreenSize().width*0.9), (int)(Toolkit.getDefaultToolkit().getScreenSize().height*0.9)));
        setExtendedState(MAXIMIZED_BOTH);
        setVisible(true);
    }

    public Workbench(JComponent workSpace) {
        this.workSpace = workSpace;
        getContentPane().setLayout(new BorderLayout());
        getContentPane().add(workSpace, BorderLayout.CENTER);

        setDefaultCloseOperation(WindowConstants.EXIT_ON_CLOSE);
        setResizable(true);
        setSize(new Dimension((int)(Toolkit.getDefaultToolkit().getScreenSize().width*0.9), (int)(Toolkit.getDefaultToolkit().getScreenSize().height*0.9)));
        setExtendedState(MAXIMIZED_BOTH);
        setVisible(true);
    }

    public void add(Toolbar toolbar, String anchor) {
        toolbar.setAnchor(anchor);
        getContentPane().add(toolbar, anchor);
        revalidate();
    }

    public void add(TabbedBar tabbedBar, String anchor) {
        getContentPane().add(tabbedBar, anchor);
        revalidate();
    }

    protected void setWorkSpace(JComponent workSpace) { this.workSpace = workSpace; }
    protected JComponent getWorkSpace() { return workSpace; }
}

File: src/henaxel/workbench/Workbench.kt
package henaxel.workbench

import java.awt.BorderLayout
import java.awt.Dimension
import java.awt.Toolkit
import javax.swing.JComponent
import javax.swing.JFrame

class Workbench(var workSpace: JComponent) : JFrame() {

    init{
        contentPane.layout = BorderLayout()
        contentPane.add(workSpace, BorderLayout.CENTER)

        defaultCloseOperation = EXIT_ON_CLOSE
        isResizable = true
        size = Dimension(
            (Toolkit.getDefaultToolkit().screenSize.width * 0.9).toInt(),
            (Toolkit.getDefaultToolkit().screenSize.height * 0.9).toInt()
        )
        extendedState = MAXIMIZED_BOTH
        isVisible = true
    }

    fun add(toolbar: Toolbar, anchor: String) {
        toolbar.setAnchor(anchor)
        contentPane.add(toolbar, anchor)
        revalidate()
    }
}